### ISSUE TITLE
DisableExecuteApiEndpoint not working with OpenApiVersion>=3.x.x

### ISSUE THREAD

**Issue by forzagreen** [aws/serverless-application-model#2396]
### Description:
`DisableExecuteApiEndpoint: True` is not working (i.e. the default endpoint is not disabled) when `OpenApiVersion` is set to 3.x.x versions (tested with `3.0.1` and `3.0.3`).

It works correctly when `OpenApiVersion: "2.0"`.

My resource:
```yaml
Resources:
  Api:
    Type: AWS::Serverless::Api
    Properties:
      StageName: MyStage
      OpenApiVersion: "3.0.3"
      DisableExecuteApiEndpoint: True
```

When deployed, the processed template contains:
```json
"openapi": "3.0.3",
"x-amazon-apigateway-endpoint-configuration": {
  "disableExecuteApiEndpoint": true
}
```
But still the API Gateway default endpoint is not disabled.

I think this is a CloudFormation bug.

### Workaround: `OpenApiVersion: "2.0"`
It works in these 2 cases:
- When I set `OpenApiVersion: "2.0"`
- If I don't add `OpenApiVersion` at all. The translated JSON template gets `"swagger": "2.0"`.
  - But I need to pass `OpenApiVersion` to prevent SAM from creating a default stage `Stage` (cf. https://github.com/aws/serverless-application-model/issues/2077 and https://github.com/aws/serverless-application-model/issues/191) 

The processed template with `OpenApiVersion: "2.0" has:
```json
"swagger": "2.0",
"x-amazon-apigateway-endpoint-configuration": {
  "disableExecuteApiEndpoint": true
}
```

### Packages versions:
```
aws-lambda-builders==1.16.0
aws-sam-cli==1.49.0
aws-sam-translator==1.45.0
```

**Comment by jfuss** [comment]
@forzagreen Thanks for the report. Upon first look, this appears to be a SAM Spec issue in how we translate. The API Gateway [docs](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-swagger-extensions-endpoint-configuration.html) mention that "For OpenAPI 3.0, it should be present under the vendor extensions of the [Server object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#serverObject)." But in 2.0: "This extension is an extended property of the [OpenAPI Operation](https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#operationObject) object. This object should be present in [top-level vendor extensions](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#vendorExtensions) for Swagger 2.0".

Looking at the code, `AWS::Serverless::Api` does not consider the OpenApiVersion at all: https://github.com/aws/serverless-application-model/blob/bdbe4127244b7ec16c98d4aebcc58d77679ead40/samtranslator/model/api/api_generator.py#L317. That line calls this function: https://github.com/aws/serverless-application-model/blob/bdbe4127244b7ec16c98d4aebcc58d77679ead40/samtranslator/swagger/swagger.py#L128 and just added it as if 2.0 was used. It seems like an assumption was made that `AWS::Serverless::Api` and therefore API Gateway RestApis only support 2.0 swagger, which is not the case. 

Marking this as a bug. This might be more challenging to support if we don't support Intrinsics in the `OpenApiVersion` field.

**Comment by aaythapa** [comment]
Closing this as it was resolved with #2560
